src/test/java/com/github/commonsrdf/api/AbstractRDFTermFactoryTest.java

```diff
@@ -0,0 +1,349 @@
+		assertEquals(subject.ntriplesString(), triple.getSubject()
+				.ntriplesString());
+		assertEquals(predicate.ntriplesString(), triple.getPredicate()
+				.ntriplesString());
+		assertEquals(object.ntriplesString(), triple.getObject()
+				.ntriplesString());
+
+	}
+
+	@Test
+	public void createTripleBnodeIRI() {
+		BlankNode subject;
+		IRI predicate;
+		IRI object;
+		Triple triple;
+		try {
+			subject = factory.createBlankNode("b1");
+			predicate = factory.createIRI("http://example.com/pred");
+			object = factory.createIRI("http://example.com/obj");
+			triple = factory.createTriple(subject, predicate, object);
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException(ex);
+			return;
+		}
+
+		// NOTE: We do not require object equivalence after insertion,
+		// but the ntriples should match
+		assertEquals(subject.ntriplesString(), triple.getSubject()
+				.ntriplesString());
+		assertEquals(predicate.ntriplesString(), triple.getPredicate()
+				.ntriplesString());
+		assertEquals(object.ntriplesString(), triple.getObject()
+				.ntriplesString());
+	}
+
+	@Test
+	public void createTripleBnodeTriple() {
+		BlankNode subject;
+		IRI predicate;
+		Literal object;
+		Triple triple;
+		try {
+			subject = factory.createBlankNode();
+			predicate = factory.createIRI("http://example.com/pred");
+			object = factory.createLiteral("Example", "en");
+			triple = factory.createTriple(subject, predicate, object);
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException(ex);
+			return;
+		}
+
+		// NOTE: We do not require object equivalence after insertion,
+		// but the ntriples should match
+		assertEquals(subject.ntriplesString(), triple.getSubject()
+				.ntriplesString());
+		assertEquals(predicate.ntriplesString(), triple.getPredicate()
+				.ntriplesString());
+		assertEquals(object.ntriplesString(), triple.getObject()
+				.ntriplesString());
+	}
+
+	@Before
+	public void getFactory() {
+		factory = createFactory();
+	}
+
+	@Test(expected = Exception.class)
+	public void invalidBlankNode() throws Exception {
+		try {
+			factory.createBlankNode("with:colon").ntriplesString();
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException("createBlankNode(String) not supported",
+					ex);
+			return;
+		}
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void invalidIRI() throws Exception {
+		try {
+			factory.createIRI("<no_brackets>");
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException("createIRI not supported", ex);
+			return;
+		}
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void invalidLiteralLang() throws Exception {
+		try {
+			factory.createLiteral("Example", "");
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException(
+					"createLiteral(String,String) not supported", ex);
+			return;
+		}
+	}
+
+	@Test(expected = Exception.class)
+	public void invalidTriplePredicate() {
```

Comment:
We still have an open (kind of) discussion about supporting Generalized RDF, where this would be valid, but we would probably use another interface or method to support Generalized RDF so this test is useful as is.